src/main/java/mat/client/shared/CQLLibraryResultTable.java

```diff
@@ -107,27 +76,27 @@ public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSet
             public SafeHtml getValue(CQLLibraryDataSetObject object) {
                 return CellTableUtility.getColumnToolTip(MeasureDetailsUtil.defaultTypeIfBlank(object.getLibraryModelType()));
             }
-
-            /*
-                Single Click to select row and enable checkbox
-                Double Click to navigate to CQL Composer tab
-            */
-            @Override
-            public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
-                if (object.getLastClick() < System.currentTimeMillis() - DELAY_TIME) {
-                    object.setSelected(!object.isSelected());
-                    selectionModel.setSelected(object, object.isSelected());
-                } else if (object.isFhirEditOrViewable()) {
-                    SelectionEvent.fire(fireEvent, object);
-                }
-                object.setLastClick(System.currentTimeMillis());
-            }
         };
         if (MatContext.get().getFeatureFlagStatus(FeatureFlagConstant.MAT_ON_FHIR))
             table.addColumn(model, SafeHtmlUtils.fromSafeConstant("<span title='Version'>" + "Model" + "</span>"));
 
+        table.addCellPreviewHandler(event -> {
+            String eventType = event.getNativeEvent().getType();
+            obj = event.getValue();
+            event.getNativeEvent().preventDefault();
+            if (BrowserEvents.CLICK.equalsIgnoreCase(eventType)) {
+                if(previousObj == null){
+                    clickEventsOnSelectedRows(event, selectionModel, obj, fireEvent);
+                } else if (obj.getId() == previousObj.getId()) {
+                    clickEventsOnSelectedRows(event, selectionModel, obj, fireEvent);
+                } else {
+                    previousObj.setSelected(false);
+                    clickEventsOnSelectedRows(event, selectionModel, obj, fireEvent);
```

Comment:
can pls check if this works

```
if(previousObj != null) {
      previousObj.setSelected(false);
}
clickEventsOnSelectedRows(event, selectionModel, obj, fireEvent);
```

Reply:
If previousObj is same as obj, then it isn't working. Since you are setting it false, before handling the event.